mazeRunner/src/mazeRunner.js

```diff
@@ -1,68 +1,70 @@
       if (startPointJ > -1) {
-        return [i, startPointJ];
+        return [this.maze.indexOf(point), startPointJ];
       }
     }
   };
 
-  // takes a direction (string) and position (array) and returns an array
-  const newPosition = (direction, position) => {
-    switch (direction) {
-      case "E":
-        return [position[0], position[1] + 1];
-      case "W":
-        return [position[0], position[1] - 1];
-      case "N":
-        return [position[0] - 1, position[1]];
-      case "S":
-        return [position[0] + 1, position[1]];
+  // takes a direction (string) and position (tuple) and returns new position (tuple)
+  newPosition = (direction, position) => {
+    if (Array.isArray(position) && position.length === 2) {
+      switch (direction) {
+        case "E":
+          return [position[0], position[1] + 1];
+        case "W":
+          return [position[0], position[1] - 1];
+        case "N":
+          return [position[0] - 1, position[1]];
+        case "S":
+          return [position[0] + 1, position[1]];
+      }
+    } else {
+      throw new Error("Position must be a tuple");
     }
   };
 
-  //console.log("original position of the 2", startingPosition(maze));
+  // calculates the outcome at the maze position
+  result = (position) => {
+    const mazePoint = this.maze[position[0]][position[1]];
 
-  const result = (maze, position) => {
-    // console.log("position in result", position);
-    // console.log("maze point", maze[position[0]][position[1]]);
-    if (
-      maze[position[0]][position[1]] === 0 ||
-      maze[position[0]][position[1]] === 2
-    ) {
+    if (mazePoint === 0 || mazePoint === 2) {
       return "Lost";
-    } else if (maze[position[0]][position[1]] === 1) {
+    } else if (mazePoint === 1) {
       return "Dead";
-    } else if (maze[position[0]][position[1]] === 3) {
+    } else if (mazePoint === 3) {
       return "Finish";
     }
   };
 
-  for (let i = 0; i < directions.length; i++) {
-    if (i === 0) {
-      position = startingPosition(maze);
-    }
+  attemptWith = (directions) => {
+    let position = this.startingPosition();
+    let moveNumber = 0;
 
-    position = newPosition(directions[i], position);
-    //console.log("position in for loop", position);
-    if (position.includes(maze.length) || position.includes(-1)) {
-      return "Dead";
-    }
-    outcome = result(maze, position);
-    //console.log("outcome at new position", outcome);
-    // console.log("i", i + 1);
-    // console.log("number of directions", directions.length);
-    if (outcome === "Dead" || outcome === "Finish") {
-      return outcome;
-    }
-    if (outcome === "Lost" && i + 1 === directions.length) {
-      return "Lost";
+    for (const direction of directions) {
+      position = this.newPosition(direction, position);
+      moveNumber++;
+      if (position.includes(this.maze.length) || position.includes(-1)) {
+        return "Dead";
+      }
+      let outcome = this.result(position);
```

Comment:
You can now make this variable a const instead of a let, as it isn't mutated.
```suggestion
      const outcome = this.result(position);
```

For a bit of context, it's better to have variables as `const` where possible, as when reading the code it shows the reader that no where else is mutating this variable.